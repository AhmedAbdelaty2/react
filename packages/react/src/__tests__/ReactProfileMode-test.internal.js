/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails react-core
 */

'use strict';

let React;
let ReactFeatureFlags;
let ReactTestRenderer;

describe('ReactProfileMode', () => {
  [true, false].forEach(enabled => {
    describe(`enableProfileModeMetrics feature flag ${
      enabled ? 'enabled' : 'disabled'
    }`, () => {
      beforeEach(() => {
        jest.resetModules();

        ReactFeatureFlags = require('shared/ReactFeatureFlags');
        ReactFeatureFlags.enableProfileModeMetrics = enabled;
        React = require('react');
        ReactTestRenderer = require('react-test-renderer');
      });

      it('should support an empty mode', () => {
        expect(
          ReactTestRenderer.create(<React.ProfileMode />).toJSON(),
        ).toMatchSnapshot();
        expect(
          ReactTestRenderer.create(
            <div>
              <React.ProfileMode />
            </div>,
          ).toJSON(),
        ).toMatchSnapshot();
      });

      it('should render children', () => {
        const ProfiledComponent = ({name}) => <span>{name}</span>;
        const renderer = ReactTestRenderer.create(
          <div>
            Hi
            <React.ProfileMode>
              <span>there</span>
              <ProfiledComponent name="ProfileMode" />
            </React.ProfileMode>
          </div>,
        );
        expect(renderer.toJSON()).toMatchSnapshot();
      });

      it('should support nested ProfileModes', () => {
        const ProfiledComponent = ({name}) => <div>Hi, {name}</div>;
        class ExtraProfiledComponent extends React.Component {
          render() {
            return <block>Hi, {this.props.name}</block>;
          }
        }
        const renderer = ReactTestRenderer.create(
          <React.ProfileMode>
            <ProfiledComponent name="Brian" />
            <React.ProfileMode>
              <ExtraProfiledComponent name="Brian" />
              <span>Now with extra profile strength!</span>
            </React.ProfileMode>
          </React.ProfileMode>,
        );
        expect(renderer.toJSON()).toMatchSnapshot();
      });
    });
  });
});
