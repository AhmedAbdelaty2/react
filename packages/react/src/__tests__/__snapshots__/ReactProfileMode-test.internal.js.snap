// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactProfileMode enableProfileModeMetrics feature flag disabled should render children 1`] = `
<div>
  Hi
  <span>
    there
  </span>
  <span>
    ProfileMode
  </span>
</div>
`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag disabled should support an empty mode 1`] = `null`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag disabled should support an empty mode 2`] = `<div />`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag disabled should support nested ProfileModes 1`] = `
Array [
  <div>
    Hi, 
    Brian
  </div>,
  <block>
    Hi, 
    Brian
  </block>,
  <span>
    Now with extra profile strength!
  </span>,
]
`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag enabled should render children 1`] = `
<div>
  Hi
  <span>
    there
  </span>
  <span>
    ProfileMode
  </span>
</div>
`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag enabled should support an empty mode 1`] = `null`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag enabled should support an empty mode 2`] = `<div />`;

exports[`ReactProfileMode enableProfileModeMetrics feature flag enabled should support nested ProfileModes 1`] = `
Array [
  <div>
    Hi, 
    Brian
  </div>,
  <block>
    Hi, 
    Brian
  </block>,
  <span>
    Now with extra profile strength!
  </span>,
]
`;
